draft-packages/menu/KaizenDraft/Menu/Menu.tsx

```diff
@@ -73,64 +35,21 @@ const Menu: Menu = props => {
     setIsMenuVisible(false)
   }
 
-  const { button } = props
-
-  return render({
-    ...props,
-    align,
-    autoHide,
-    isMenuVisible,
-    dropdownButtonContainer,
-    hideMenuDropdown,
-    menuButton: React.cloneElement(button, {
-      onClick: (e: any) => {
-        e.stopPropagation()
-        toggleMenuDropdown()
-      },
-      onMouseDown: (e: any) => e.preventDefault(),
-      "aria-haspopup": true,
-      "aria-expanded": isMenuVisible,
-    }),
-  })
-}
-
-type RenderProps = {
-  menuButton: React.ReactElement
-  isMenuVisible: boolean
-  dropdownButtonContainer: React.RefObject<HTMLDivElement>
-  hideMenuDropdown: () => void
-}
-
-export const render = (props: GenericMenuProps & RenderProps) => {
-  const menu = (
-    <MenuDropdown
-      position={getPosition(props.dropdownButtonContainer)}
-      align={props.align}
-      hideMenuDropdown={props.hideMenuDropdown}
-      width={props.dropdownWidth}
-      id={props.dropdownId}
-      autoHide={props.autoHide}
-    >
-      {props.children}
-    </MenuDropdown>
-  )
   return (
-    <div
-      className={styles.dropdown}
-      data-automation-id={props.automationId}
-      ref={props.dropdownButtonContainer}
-    >
-      {props.menuButton}
-      {props.isMenuVisible ? menu : null}
-    </div>
+    <StatelessMenu
+      {...rest}
+      isMenuVisible={isMenuVisible}
+      toggleMenuDropdown={toggleMenuDropdown}
+      hideMenuDropdown={hideMenuDropdown}
+      // `StatelessMenu` is one of the newer components, so it uses a render function,
+      // as opposed to `React.cloneElement`.
+      // `cloneElement` has its problems, mainly because it's somewhat magical, can lead
+      // to unexpected behaviour, and it doesn't self document. Hence, the switch
+      // to the render function. It would be nice if the `Menu` component also
+      // used this pattern, but it's probably not worth the time and effort.
```

Comment:
I didn't do the original code, but I added this comment. LMK if I made incorrect assumptions here.

Reply:
That was the general feeling about it so it's all good. From memory that `cloneElement` was itself ported from an even earlier version of the component 🙃